scala-thoughts.md

```diff
@@ -0,0 +1,8 @@
+
+It is very easy to do recursion with Scala, I like how you have to declare the type when writing a recursive function. It is also very easy to create parameters to functions in Scala.
+
+I don't love the list creation in Scala. I was originally trying to get user input and put it into a list (I later switched to putting it into a string) and I was really struggling to understand what the rules regarding lists are. I would like to learn more about how lists work in Scala, and understand how one can edit a list. However, I did find it easy to work with strings, which was very nice.
+
+I would also like to learn about the type formatting in Scala. It seems like it was made with great intention. If I had a better understanding the type formatting I could use type information to my advantage when writing and debugging code.
+
+The one thing that I did struggle with today was sbt being buggy. I was having a problem where the readLine() function wouldn't read my new line characters, and would instead read them as "^M". To solve this problem I had to close the terminal and start a new terminal. This was just frustrating as it took me some time to figure out that the problem was not my fault.
```

Comment:
I had similar issues with using sbt, but I was mainly using codespaces so I had assumed the origin of the issue was codespaces and not specific to the terminal. It is indeed frustrating, and thank you for pointing this out!